src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -300,6 +267,87 @@ Literal Literal::logical_binop_impl(std::array<std::array<TriStateBool, 3>, 3> c
     return Literal::make<datatypes::xsd::Boolean>(res == TriStateBool::True, node_storage);
 }
 
+std::partial_ordering Literal::compare_impl(Literal const &other, std::partial_ordering *out_type_ordering) const {
+    using datatypes::registry::DatatypeRegistry;
+
+    if (this->handle_.null() || other.handle_.null()) {
+        return this->handle_ == other.handle_
+                       ? std::partial_ordering::equivalent
+                       : std::partial_ordering::unordered;
+    }
+
+    auto const this_datatype = this->datatype().identifier();
+    auto const this_entry = DatatypeRegistry::get_entry(this_datatype);
+    assert(this_entry != nullptr);
+
+    if (this_entry->compare_fptr == nullptr) {
+        return std::partial_ordering::unordered; // not comparable
+    }
+
+    auto const other_datatype = other.datatype().identifier();
+
+    auto const datatype_cmp_res = this_datatype <=> other_datatype;
```

Comment:
```suggestion
    std::strong_ordering const datatype_cmp_res = this_datatype <=> other_datatype;
```

We are comparing two std::string_views. Result is a [`std::strong_ordering`](https://en.cppreference.com/w/cpp/string/basic_string_view/operator_cmp#Notes). I would suggest to add the type explicitly here. It helps to understand why `compare_with_extensions` is `std::string_ordering`.